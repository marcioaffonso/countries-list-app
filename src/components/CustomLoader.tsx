import React from 'react';
import { View, ActivityIndicator, StyleSheet } from 'react-native';

export default function CustomLoader() {
  return (
    <View style={styles.loaderContainer}>
      <ActivityIndicator size={'large'} />
    </View>
  );
};

const styles = StyleSheet.create({
  loaderContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center'
  }
});
