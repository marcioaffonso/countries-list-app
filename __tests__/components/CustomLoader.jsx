import React from 'react';
import renderer from 'react-test-renderer';
import { expect, test } from '@jest/globals';
import CustomLoader from '../../src/components/CustomLoader';

test('renders loader correctly', () => {
  const tree = renderer.create(<CustomLoader />).toJSON();
  expect(tree).toMatchSnapshot();
});
