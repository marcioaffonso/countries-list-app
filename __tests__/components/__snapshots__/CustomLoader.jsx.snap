// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loader correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    size="large"
  />
</View>
`;
